### PR TITLE
Add robust RMI implementation

### PR DESCRIPTION
**Pull request by alhuan**
Add the robust RMI implementation to replace the C++ RMI implementation. 

Submodule analysis-rmi updates should fix existing build errors that started from #30.

### CODE REVIEW

File: competitors/rmi_cpp_robust.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ public:
+  uint64_t Build(const std::vector<KeyValue<KeyType>>& data) {
+    std::vector<KeyType> loading_data;
+    loading_data.reserve(data.size());
+    for (auto& itm : data) {
+      loading_data.push_back(itm.key);
+    }
+
+    return util::timing([&] {
+      auto rmi_ptr_one = std::make_unique<RMIType<KeyType, Layer1, Layer2>>(
```
Comment (andreaskipf) [inline competitors/rmi_cpp_robust.h]: Let's build them one by one to save memory.

File: benchmarks/benchmark_rmi_alt.cc [hunk 2]
```diff
@@ -3,53 +3,81 @@
...
+    benchmark.template Run<
+        RMICppRobust<uint32_t, rmi::LinearRegression, rmi::LinearRegression,
+                     rmi::RmiLAbsRobust, 524288, 7>>();
+    benchmark.template Run<
+        RMICppRobust<uint32_t, rmi::LinearRegression, rmi::LinearRegression,
+                     rmi::RmiLAbsRobust, 2097152, 8>>();
+    benchmark.template Run<
+        RMICppRobust<uint32_t, rmi::LinearRegression, rmi::LinearRegression,
+                     rmi::RmiLAbsRobust, 8388608, 9>>();
+    benchmark.template Run<
```
Comment (andreaskipf) [inline benchmarks/benchmark_rmi_alt.cc:40]: Why did we change the model parameters?

**Review by andreaskipf — CHANGES_REQUESTED** [review]

File: competitors/rmi_cpp_robust.h (hunk 1 above)
Comment (alhuan) [reply competitors/rmi_cpp_robust.h]: Done

File: benchmarks/benchmark_rmi_alt.cc (hunk 2 above)
Comment (alhuan) [reply benchmarks/benchmark_rmi_alt.cc:40]: The old parameters were still 4 orders of magnitude smaller than the largest reference RMIs, which seemed suboptimal for comparison

**Review by andreaskipf — APPROVED** [review]
Looks good!

**Pull request merged**